### ISSUE TITLE
Questions about model evaluation parts

### ISSUE THREAD

**Issue by SeYoungKim** [trinhan/crc-ihc-classification#4]
Hello

I'm trying to understand the code. And... I have questions about model evaluation parts...

About evaluating RF classifier with the training data...

As I understand, the Random Forest classifier is trained by using the whole AMC dataset. So AMC dataset is used as a training set and a set of the other data(LUMC, CAIRO, CAIRO2) is used as a validation set.

So when the classifier was trained with AMC dataset, its OOB error rate was 19.83%. 

And in the paper, Fig 1. C says its accuracy is 87.14%. 

![image](https://user-images.githubusercontent.com/12039768/39964747-b91feabc-56c6-11e8-831d-93a0567563f9.png)

So I looked for the part in the [4.1.1 Training the classifier, Supplementary Info...]. There was a code snippet for drawing the table(but I couldn't find it in the current repo.. is there a different source file which is same as [Supplementary Info...]?)

I guess the result of the gold-standard transcriptome-based classifier means the GE result and... isn't it the Class column values in AMCclinical.csv? 

![image](https://user-images.githubusercontent.com/12039768/39964520-52b9fdd0-56c1-11e8-9975-2d4f5dd626aa.png)

Seems like the labels used for training were used for evaluation...

Is it correct that 87.14% is the result of evaluation with the training data? 

If so, isn't it that only the OOB value has meaning?

Thank you

**Comment by SeYoungKim** [comment]
Isn't the accuracy of the RF on the training set(per sample, not per patient) is (almost)100%?

I tried this again in the different environment with the same data and I got similar answers...

About 20% of oob error and almost 100% of accuracy on the training data..!

![image](https://user-images.githubusercontent.com/12039768/39968142-6abf3aac-5703-11e8-8aaf-3024d3c1345e.png)

**Comment by SeYoungKim** [comment]
I have one more question about evaluation...

There is a histogram like below in [Supplementary Information...] and I wonder if this graph is for test set accuracies in k-fold cross validation...

![image](https://user-images.githubusercontent.com/12039768/39966524-2bfa66c2-56e8-11e8-957b-3cf053e43747.png)

I'm reading this...

![image](https://user-images.githubusercontent.com/12039768/39966515-ff32a8e8-56e7-11e8-83cd-4ff81ec0cfc3.png)
![image](https://user-images.githubusercontent.com/12039768/39966516-02e8e204-56e8-11e8-81a2-011766550694.png)

As I understand, you're performing k-fold cross validation... 

and this table (rfpred) is the result from the one time evaluation.(70%)

![image](https://user-images.githubusercontent.com/12039768/39966562-e490b07e-56e8-11e8-8fa4-27accbff33ef.png)



And there is a picture of some distribution like below...

![image](https://user-images.githubusercontent.com/12039768/39966524-2bfa66c2-56e8-11e8-957b-3cf053e43747.png)


**My question is, Is this distribution is for the OOB distribution or k-fold cross validation...?(200 times)**

The code drawing the histogram (Core classification accuracy) is using the result from the ClassifyPrediction and I couldn't find it from the current repo... 

There are two lines: "The accuracy for the above is 0.7. This is repeated 200 times" and "Using this method, the OOB distribution and prediction accuracy is as follows". Here I can't make out which one the histogram is for...

![image](https://user-images.githubusercontent.com/12039768/39966578-3fdb0b5a-56e9-11e8-8321-1ccb52d82fe8.png)
